src/EditorFeatures/Core/CommandHandlers/GoToAdjacentMemberCommandHandler.cs

```diff
@@ -60,32 +60,35 @@ public bool ExecuteCommand(GoToPreviousMemberCommandArgs args, CommandExecutionC
 
         private VSCommanding.CommandState GetCommandStateImpl(EditorCommandArgs args)
         {
-            var document = args.SubjectBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
             var caretPoint = args.TextView.GetCaretPoint(args.SubjectBuffer);
-            return IsAvailable(document, caretPoint) ? VSCommanding.CommandState.Available : VSCommanding.CommandState.Unspecified;
-        }
-
-        private static bool IsAvailable(Document document, SnapshotPoint? caretPoint)
-        {
-            if (document?.SupportsSyntaxTree != true)
+            if (!IsAvailable(caretPoint, args.SubjectBuffer))
             {
-                return false;
+                return VSCommanding.CommandState.Unspecified;
             }
 
-            if (!caretPoint.HasValue)
+            var document = args.SubjectBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
+            if (document?.SupportsSyntaxTree != true)
             {
-                return false;
+                return VSCommanding.CommandState.Unspecified;
             }
 
-            var documentSupportsFeatureService = document.Project.Solution.Workspace.Services.GetService<IDocumentSupportsFeatureService>();
-            return documentSupportsFeatureService?.SupportsNavigationToAnyPosition(document) == true;
+            return VSCommanding.CommandState.Available;
         }
 
+        private static bool IsAvailable(SnapshotPoint? caretPoint, ITextBuffer subjectBuffer) =>
```

Comment:
>vailable(SnapshotPoint? caretPoint, ITextBuffer subjectBuffer) [](start = 31, length = 62)

This method is a bit odd. It does two unrelated checks. I think it would be better to just inline it to its call site.